### ISSUE TITLE
Avoid repeating default values for super parameters

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#3338]
If the default values for super parameters in a subclass are the same as in the super class, they don't have to be repeated as they will be copied from the super constructor. A lint could remind people that the repetition is unnecessary to keep it DRY.

**Examples**

Bad:

```dart
class Bar extends Foo {
  Bar({super.value = true}); // Lint on this line that "= true" isn't necessary.
}

class Foo {
  Foo({this.value = true});
  final bool value;
}
```

Good:

```dart
class Bar extends Foo {
  Bar({super.value}); 
}

class Foo {
  Foo({this.value = true});
  final bool value;
}
```

Also ok:

```dart
class Bar extends Foo {
  Bar({super.value = false}); // Don't lint because the default was changed.
}

class Foo {
  Foo({this.value = true});
  final bool value;
}
```

**Additional context**
Rules about copying the default value from the associated super-constructor: https://github.com/dart-lang/language/blob/master/working/1855%20-%20super%20parameters/proposal.md#parameter-types-and-default-values

**Comment by pq** [comment]
I wonder if extending [`avoid_redundant_argument_values`](https://dart-lang.github.io/linter/lints/avoid_redundant_argument_values.html) to handle the super parameter case would be sufficient?

**Comment by pq** [comment]
Looking a bit closer, this lint currently is only about _invocations_ and leaves redundancy in **declarations** well enough alone.

If we want to extend it, should we consider cases like this too?


```dart
class A {
  int a = 0;
  A([this.a = 0]);
}
```

**Comment by goderbauer** [comment]
I think it would make sense to split invocations and declarations into separate lints (not feeling super strongly about this, though). The declaration one could also include the case you mentioned above.

**Comment by pq** [comment]
I'm leaning that way too.  Maybe:

* `avoid_redundant_default_values`, or
* `no_redundant_default_values`

**Comment by bwilkerson** [comment]
I agree that a separate lint is better, and I like the second suggested name better.

**Comment by pq** [comment]
Closing this in favor of a proposal in #3343.

Thanks @goderbauer!
